dialog.cpp

```diff
@@ -160,27 +160,44 @@ void Dialog::readPacmanLogFile(const QString &logFile)
 
     QStringList names;
 
-    const QRegExp rx("\\[([0-9-]+)T*\\s*\\d+:\\d+:*\\d*\\+*\\d*\\]\\s\\[(PACMAN|ALPM)\\]\\s(installed|removed|upgraded)\\s([\\w-]+)\\s\\((.+)\\)");
+    const QRegExp rx("\\[(.+)\\]\\s\\[(PACMAN|ALPM|PACKAGEKIT)\\]\\s(installed|removed|upgraded)\\s([\\w-]+)\\s\\((.+)\\)");
 
     while(!file.atEnd()) {
         QString line = file.readLine();
         
-        if(line.contains("[PACKAGEKIT]"))
-            continue;
-        
         rx.indexIn(line);
-        if(rx.cap(3).isEmpty())
+        if(rx.cap(0).isEmpty())
             continue;
 
-        QDate date = QDate::fromString(rx.cap(1), "yyyy-MM-dd");
+        QString timestamp = rx.cap(1);
+        QString who = rx.cap(2);
+        QString op = rx.cap(3);
+        QString pkg = rx.cap(4);
+        QString ver = rx.cap(5);
+
+        int T_ix = timestamp.indexOf("T");
+        if (T_ix != -1){
+            // New-style timestamp yyyy-MM-ddThh:mm:ss-zzzz. Strip offset and replace
+            // the T with a space:
+            timestamp.replace(T_ix, 1, " ");
+            timestamp.truncate(timestamp.lastIndexOf('-'));
```

Comment:
This should also consider timestamps of the form `yyyy-MM-ddThh:mm:ss+zzzz`.

(cf https://github.com/gcala/pacmanlogviewer/issues/3#issuecomment-565857321)

Reply:
Ah, thanks. Will fix.